nf_core/modules/module_test.py

```diff
@@ -0,0 +1,145 @@
+                raise UserWarning(
+                    "Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
+                )
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        module_dir = Path("modules") / self.module_name
+
+        # First, sanity check that the module directory exists
+        if not module_dir.is_dir():
+            raise UserWarning(
+                f"Cannot find directory '{module_dir}'. Should be TOOL/SUBTOOL or TOOL. Are you running the tests inside the nf-core/modules main directory?"
+            )
+
+    def _set_profile(self):
+        """Set $PROFILE env variable.
+        The config expects $PROFILE and Nextflow fails if it's not set.
+        """
+        if os.environ.get("PROFILE") is None:
+            os.environ["PROFILE"] = ""
+            if self.no_prompts:
+                log.info(
+                    "Setting environment variable '$PROFILE' to an empty string as not set.\n"
+                    "Tests will run with Docker by default. "
+                    "To use Singularity set 'export PROFILE=singularity' in your shell before running this command."
+                )
+            else:
+                question = {
+                    "type": "list",
+                    "name": "profile",
+                    "message": "Choose software profile",
+                    "choices": ["Docker", "Singularity", "Conda"],
+                }
+                answer = questionary.unsafe_prompt([question], style=nf_core.utils.nfcore_question_style)
+                profile = answer["profile"].lower()
+                os.environ["PROFILE"] = profile
+                log.info(f"Setting environment variable '$PROFILE' to '{profile}'")
+
+    def _check_profile(self):
+        """Check if profile is available"""
+        profile = os.environ.get("PROFILE")
+        # Make sure the profile read from the environment is a valid Nextflow profile.
+        valid_nextflow_profiles = ["docker", "singularity", "podman", "shifter", "charliecloud", "conda"]
+        if profile in valid_nextflow_profiles:
+            if not which(profile):
+                raise UserWarning(f"The PROFILE '{profile}' set in the shell environment is not available.")
+        else:
+            raise UserWarning(
+                f"The PROFILE '{profile}' set in the shell environment is not valid.\n"
+                f"Valid Nextflow profiles are {valid_nextflow_profiles}."
+            )
+
+    def _run_pytests(self):
+        """Given a module name, run tests."""
+        # Print nice divider line
+        console = rich.console.Console()
+        console.print("[black]" + "─" * console.width)
```

Comment:
https://rich.readthedocs.io/en/stable/console.html#rules

```suggestion
        console.rule(self.module_name, style="black")
```